rwiki/rwiki-tool/tool/src/java/uk/ac/cam/caret/sakai/rwiki/tool/VelocityInlineDispatcher.java

```diff
@@ -96,11 +101,22 @@ public void dispatch(String path, HttpServletRequest request,
 		// EventCartridge ec = new EventCartridge();
 		// ec.addEventHandler(new ExcludeEscapeHtmlReference());
 		// ec.attachToContext(vcontext);
+		RequestScopeSuperBean requestScopeSuperBean =  RequestScopeSuperBean.getFromRequest(request);
 
 		vcontext.put("session", request.getSession());
 		vcontext.put("request", request);
-		vcontext.put("requestScope", RequestScopeSuperBean.getFromRequest(request));
+		vcontext.put("requestScope", requestScopeSuperBean);
+		vcontext.put("groups", (requestScopeSuperBean.getHomeBean().getHomeLinkValue().equals(requestScopeSuperBean.getViewBean().getLocalName()))? "" : requestScopeSuperBean.getPageGroups());
```

Comment:
Let's do it simpler.

String localPage = requestScopeSuperBean.getViewBean().getLocalName();
String homePage= requestScopeSuperBean.getHomeBean().getHomeLinkValue();
String groups = "";
if (!StringUtils.equalsIgnoreCase(homePage, localPage)) {
groups  = requestScopeSuperBean.getPageGroups();
}
vcontext.put("groups", groups);

WDYT?

Reply:
Ok. For me look good